src/replacer.rs

```diff
@@ -28,6 +28,19 @@ pub fn replace_strings(rules: &Rules, raw: &str) -> String {
         }
     }
 
+    // regex replacements
+    for regex_replacement in rules.regex_replacement_list.iter() {
+        if Value::as_array(regex_replacement).unwrap().len() == 3 {
```

Comment:
This made me wonder if this implementation should go further than just with 3 values. Initially I thought such a regex implementation would only take two arguments and basically work like the `replace_all` function. But thinking about it, I can absolutely see why 3 arguments can be even more helpful, though many use cases could also be covered by named capture groups (but not all!).

Would you be interested in implementing a second form of this that accepts two arguments and replaces every matched occurrence with that string? This of course could be done outside this PR as a follow-up.